craft_parts/state_manager/build_state.py

```diff
@@ -16,23 +16,25 @@
 
 """State definitions for the build step."""
 
-from typing import Any, Dict, List, Optional
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
 
 from overrides import override
-from pydantic import validator
+from pydantic import AfterValidator
+from typing_extensions import Annotated
 
 from .step_state import StepState, validate_hex_string
 
+if TYPE_CHECKING:
+    HexString = str
+else:
+    HexString = Annotated[str, AfterValidator(validate_hex_string)]
```

Comment:
```suggestion
HexString = Annotated[str, AfterValidator(validate_hex_string)]
```

seems to work fine on my machine

Reply:
done